### ISSUE TITLE
Missing rrd?

### ISSUE THREAD

**Issue by ghost** [ntop/ntopng#1232]
![screen shot 2017-05-12 at 14 31 30](https://cloud.githubusercontent.com/assets/4884243/25998161/f4c8736c-371f-11e7-8769-3bd7c2a0cc9d.png)

**Comment by emanuele-f** [comment]
What ntopng version are you using (see `ntopng -V`)? Can you post the ntopng log from the terminal?

**Comment by ghost** [comment]
v.2.5.170502 [Enterprise/Professional/Embedded Edition]
GIT rev:   dev:8537b2cc0a6d237b910871b6a6d21b4242fa6514:20170502
Pro rev:   r1055
System Id: FD66D271499602D2
Built on:  Raspbian GNU/Linux 8.0 (jessie)

Apparently nothing in the log….

drwxr-xr-x 2 root root  4096 May 15 06:25 .
drwxr-xr-x 9 root root  4096 May 15 06:25 ..
-rwxrwxrwx 1 root root     0 May 15 06:25 ntopng.log
-rwxrwxrwx 1 root root     0 May 15 06:25 ntopng.log.1
-rwxrwxrwx 1 root root     0 May 14 06:25 ntopng.log.2
-rwxrwxrwx 1 root root 29968 May 13 06:25 ntopng.log.3

**Comment by emanuele-f** [comment]
What's the output of `ls -l /var/tmp/ntopng/0/rrd/all`?
In order to get the full log, please stop ntopng `systemctl stop ntopng` ad run it from terminal `ntopng /etc/ntopng.conf`

**Comment by ghost** [comment]
`ls: cannot access /var/tmp/ntopng/0/rrd/all: No such file or directory`

```
15/May/2017 10:10:47 [Prefs.cpp:897] All HTTP user login disabled
15/May/2017 10:10:47 [Prefs.cpp:1129] Logging into /var/tmp/ntopng/ntopng.log
15/May/2017 10:10:47 [Ntop.cpp:1327] Setting local networks to 127.0.0.0/8
15/May/2017 10:10:47 [Redis.cpp:109] Successfully connected to redis 127.0.0.1:6379@0
15/May/2017 10:10:47 [NtopPro.cpp:143] [LICENSE] Read license from Redis []
15/May/2017 10:10:47 [NtopPro.cpp:195] ERROR: [LICENSE] Invalid or missing ntopng License [Empty license file]
15/May/2017 10:10:47 [NtopPro.cpp:208] WARNING: [LICENSE] ntopng will now run in enterprise edition for 10 minutes
15/May/2017 10:10:47 [NtopPro.cpp:210] WARNING: [LICENSE] before returning to community mode
15/May/2017 10:10:47 [NtopPro.cpp:211] WARNING: [LICENSE] You can buy a permanent license at http://shop.ntop.org
15/May/2017 10:10:47 [NtopPro.cpp:212] WARNING: [LICENSE] or run ntopng in community mode starting
15/May/2017 10:10:47 [NtopPro.cpp:213] WARNING: [LICENSE] ntopng --community
15/May/2017 10:10:47 [Ntop.cpp:1301] Parent process is exiting (this is normal)
```

**Comment by emanuele-f** [comment]
Please remove the `-e` option from the configuration file and retry. Kill existing ntopng instances with `killall ntopng` before retrying

**Comment by ghost** [comment]
HTTP working, HTTPS not.

```
15/May/2017 10:39:33 [Prefs.cpp:897] All HTTP user login disabled
15/May/2017 10:39:33 [Ntop.cpp:1327] Setting local networks to 127.0.0.0/8
15/May/2017 10:39:33 [Redis.cpp:109] Successfully connected to redis 127.0.0.1:6379@0
15/May/2017 10:39:33 [NtopPro.cpp:143] [LICENSE] Read license from Redis []
15/May/2017 10:39:33 [NtopPro.cpp:195] ERROR: [LICENSE] Invalid or missing ntopng License [Empty license file]
15/May/2017 10:39:33 [NtopPro.cpp:208] WARNING: [LICENSE] ntopng will now run in enterprise edition for 10 minutes
15/May/2017 10:39:33 [NtopPro.cpp:210] WARNING: [LICENSE] before returning to community mode
15/May/2017 10:39:33 [NtopPro.cpp:211] WARNING: [LICENSE] You can buy a permanent license at http://shop.ntop.org
15/May/2017 10:39:33 [NtopPro.cpp:212] WARNING: [LICENSE] or run ntopng in community mode starting
15/May/2017 10:39:33 [NtopPro.cpp:213] WARNING: [LICENSE] ntopng --community
Cannot change tcp-segmentation-offload
15/May/2017 10:39:33 [NetworkInterface.cpp:224] Executing ethtool -K wlan0 gro off gso off tso off
15/May/2017 10:39:33 [PcapInterface.cpp:86] Reading packets from interface wlan0...
15/May/2017 10:39:33 [Ntop.cpp:1432] Registered interface wlan0 [id: 0]
15/May/2017 10:39:33 [main.cpp:250] PID stored in file /var/tmp/ntopng.pid
15/May/2017 10:39:33 [Utils.cpp:368] User changed to nobody
15/May/2017 10:39:34 [HTTPserver.cpp:796] Web server dirs [/usr/share/ntopng/httpdocs][/usr/share/ntopng/scripts]
15/May/2017 10:39:34 [HTTPserver.cpp:799] HTTP server listening on port(s) 3000,127.0.0.1:3001s
15/May/2017 10:39:34 [HTTPserver.cpp:803] HTTPS server listening on port 3001
15/May/2017 10:39:34 [main.cpp:311] Working directory: /var/tmp/ntopng
15/May/2017 10:39:34 [main.cpp:313] Scripts/HTML pages directory: /usr/share/ntopng
15/May/2017 10:39:34 [Ntop.cpp:292] Welcome to ntopng armv7l v.2.5.170502 - (C) 1998-17 ntop.org
15/May/2017 10:39:34 [Ntop.cpp:297] Built on Raspbian GNU/Linux 8.0 (jessie)
15/May/2017 10:39:34 [PeriodicActivities.cpp:56] Started periodic activities loop...
15/May/2017 10:39:36 [NtopPro.cpp:272] [LICENSE] ntopng systemId: FD66D271499602D2
15/May/2017 10:39:36 [NtopPro.cpp:277] [LICENSE] ntopng is starting in demo mode
15/May/2017 10:39:36 [Ntop.cpp:565] Adding 192.168.1.14/32 as IPv4 interface address for wlan0
15/May/2017 10:39:36 [Ntop.cpp:573] Adding 192.168.1.0/24 as IPv4 local network for wlan0
15/May/2017 10:39:36 [Ntop.cpp:592] Adding fe80::6ad3:dc64:601a:108d/128 as IPv6 interface address for wlan0
15/May/2017 10:39:36 [Ntop.cpp:601] Adding fe80::6ad3:dc64:601a:108d/64 as IPv6 local network for wlan0
15/May/2017 10:39:36 [Ntop.cpp:612] Local Interface Addresses (System Host)
15/May/2017 10:39:36 [AddressTree.cpp:171] [AddressTree] 192.168.1.14/32
15/May/2017 10:39:36 [AddressTree.cpp:171] [AddressTree] fe80::6ad3:dc64:601a:108d/128
15/May/2017 10:39:36 [Ntop.cpp:614] Local Networks
15/May/2017 10:39:36 [AddressTree.cpp:171] [AddressTree] 127.0.0.0/8
15/May/2017 10:39:36 [AddressTree.cpp:171] [AddressTree] 192.168.1.0/24
15/May/2017 10:39:36 [AddressTree.cpp:171] [AddressTree] fe80::6ad3:dc64:601a:108d/64
15/May/2017 10:39:36 [NetworkInterface.cpp:1988] Started packet polling on interface wlan0 [id: 0]...
```

**Comment by ghost** [comment]
Changed -W=:3001 to -W=3001. Server takes too long to respond when accessing over HTTPS. HTTP still ok.

**Comment by emanuele-f** [comment]
If you've solved issue #1231, please close it.
About this issue, do you still see the "Missing rrd" error?

**Comment by ghost** [comment]
ntopng is now in 10 minute professional mode and i see the RRD error, but also the graph.
<img width="1418" alt="screen shot 2017-05-15 at 10 52 49" src="https://cloud.githubusercontent.com/assets/4884243/26049848/c7a71354-395c-11e7-879f-6d4610eb71bd.png">
On the console i get the following error:

`15/May/2017 10:52:17 [Lua.cpp:6503] WARNING: Script failure [/usr/share/ntopng/scripts/lua/flows_stats.lua][/usr/share/ntopng/scripts/lua/flows_stats.lua:157: attempt to concatenate a nil value]`

Now in community mode:

<img width="1427" alt="screen shot 2017-05-15 at 10 55 17" src="https://cloud.githubusercontent.com/assets/4884243/26049912/0b855c70-395d-11e7-8847-8263d2c93bba.png">

**Comment by emanuele-f** [comment]
We will do some tests and let you know, thank you.

**Comment by emanuele-f** [comment]
Can you explain the steps to get to that page? Are you using a modified version of ntopng?

**Comment by ghost** [comment]
No modified version of ntopng. I'm not smart enough for that. ;-)

To reproduce:

Homepage, Hosts, Hosts

<img width="268" alt="screen shot 2017-05-15 at 12 41 54" src="https://cloud.githubusercontent.com/assets/4884243/26054225/1794d838-396c-11e7-8962-62d09838f15c.png">

Any IPV6 host (maybe it behaves the same with IPV4, haven't tried).

<img width="361" alt="screen shot 2017-05-15 at 12 42 10" src="https://cloud.githubusercontent.com/assets/4884243/26054254/28feaca2-396c-11e7-8e02-80d6ea8bd323.png">

Click the network address link.

<img width="1120" alt="screen shot 2017-05-15 at 12 42 30" src="https://cloud.githubusercontent.com/assets/4884243/26054275/38588c36-396c-11e7-84b6-42f29275f21c.png">

An almost empty page shows up. Click the graph icon right to the word Flows.

<img width="666" alt="screen shot 2017-05-15 at 12 42 44" src="https://cloud.githubusercontent.com/assets/4884243/26054302/5403d65c-396c-11e7-962c-dc6639a7c402.png">

Professional mode.

<img width="932" alt="screen shot 2017-05-15 at 12 43 10" src="https://cloud.githubusercontent.com/assets/4884243/26054328/6f79ca72-396c-11e7-8166-7d16305fa152.png">

Community mode.

<img width="1432" alt="screen shot 2017-05-15 at 12 46 40" src="https://cloud.githubusercontent.com/assets/4884243/26054357/928bc5c4-396c-11e7-8160-028e0394f868.png">

**Comment by emanuele-f** [comment]
That was an old link pointing to the wrong page. I've fixed it with commit 3be93aad69a39f906087111b85a3d4502826e22f It will be available to you as soon an a new raspberry package will be built. Thank you for reporting.
